opcua-client/src/session/mod.rs

```diff
@@ -430,4 +431,59 @@ impl Session {
             )),
         )
     }
+
+    ///
+    /// Return namespace array from server
+    pub async fn get_namespace_array(&self) -> Result<Vec<String>, Error> {
+        let nodeid: NodeId = VariableId::Server_NamespaceArray.into();
+        let result = self
+            .read(
+                &[ReadValueId::from(nodeid)],
+                TimestampsToReturn::Neither,
+                0.0,
+            )
+            .await
+            .map_err(|status_code| {
+                Error::new(status_code, "Reading Server namespace array failed")
+            })?;
+        if let Some(Variant::Array(array)) = &result[0].value {
+            let arr = array
+                .values
+                .iter()
+                .map(|v| {
+                    //TODO iterator can handle result itself!!!
+                    if let Variant::String(s) = v {
+                        s.value().clone().unwrap_or(String::new())
+                    } else {
+                        String::new()
+                    }
+                })
+                .collect();
+            Ok(arr)
+        } else {
+            Err(Error::new(
+                StatusCode::BadNoValue,
+                format!(
+                    "Server namespace array is None. The server has an issue {:?}",
+                    result
+                ),
+            ))
+        }
+    }
+
+    ///
+    /// read namespace array from server and look for supplied namespace url
+    pub async fn get_namespace_idx(&self, url: &str) -> Result<u16, Error> {
+        let array = self.get_namespace_array().await?;
+        let idx = array.iter().position(|s| s == url).ok_or_else(|| {
+            Error::new(
```

Comment:
returning an opcua status code error here. that fits well for read_namespace_arry, but maybe not great here. What do you want @einarmo ? use thiserror?